backend/src/test/java/de/bund/digitalservice/ris/caselaw/adapter/DatabasePatchMapperServiceTest.java

```diff
@@ -0,0 +1,110 @@
+import de.bund.digitalservice.ris.caselaw.domain.User;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.UUID;
+import org.jsoup.nodes.Element;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.http.HttpHeaders;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+@ExtendWith(SpringExtension.class)
+class DatabasePatchMapperServiceTest {
+
+  private AttachmentService attachmentService;
+  private DatabasePatchMapperService service;
+
+  private static final String BASE64_IMAGE =
+      "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAUA"
+          + "AAAFCAYAAACNbyblAAAAHElEQVQI12P4"
+          + "//8/w38GIAXDIBKE0DHxgljNBAAO"
+          + "9TXL0Y4OHwAAAABJRU5ErkJggg==";
+
+  public static Element getDefaultImageTag() {
+    return new Element("img")
+        .addClass("inline")
+        .addClass("align-baseline")
+        .attr("src", BASE64_IMAGE)
+        .attr("alt", "Smallest base64 image")
+        .attr("width", "82")
+        .attr("height", "80");
+  }
+
+  @BeforeEach
+  void setup() {
+    attachmentService = mock(AttachmentService.class);
+    DatabaseDocumentationUnitPatchRepository repository =
+        mock(DatabaseDocumentationUnitPatchRepository.class);
+    service = new DatabasePatchMapperService(new ObjectMapper(), repository, attachmentService);
+  }
+
+  @Test
+  void extractAndStoreBase64Images_shouldExtractAndReplaceBase64ImageWithApiUrl() {
+    String html = "<p>Text</p>" + getDefaultImageTag().outerHtml() + "</p>";
+    JsonPatch patch = new JsonPatch(List.of(new ReplaceOperation("/foo", new TextNode(html))));
+
+    DocumentationUnit docUnit =
+        Decision.builder().uuid(UUID.randomUUID()).documentNumber("YYTestDoc0001").build();
+
+    Attachment attachment =
+        Attachment.builder().name("2e7e7908-faa3-4aca-a0ff-4bfa4dcf316a.png").build();
+
+    when(attachmentService.attachFileToDocumentationUnit(
+            any(), any(ByteBuffer.class), any(HttpHeaders.class), any()))
+        .thenReturn(attachment);
+
+    JsonPatch result = service.extractAndStoreBase64Images(patch, docUnit, User.builder().build());
+
+    assertThat(result.getOperations()).hasSize(1);
+    JsonPatchOperation op = result.getOperations().get(0);
+    assertThat(op).isInstanceOf(ReplaceOperation.class);
+
+    String newHtml = ((PathValueOperation) op).getValue().textValue();
+
+    assertThat(newHtml).doesNotContain("data:image");
+    assertThat(newHtml).doesNotContain("data:image");
+
+    assertEquals(
+        "<p>Text</p><img class=\"inline align-baseline\" src=\"/api/v1/caselaw/documentunits/YYTestDoc0001/image/2e7e7908-faa3-4aca-a0ff-4bfa4dcf316a.png\" alt=\"Smallest base64 image\" width=\"82\" height=\"80\"><p></p>",
+        newHtml);
+  }
+
+  @Test
+  void extractAndStoreBase64Images_shouldKeepOperationsWithoutImages() {
```

Comment:
We could introduce a parametrized test and add more cases (e.g. an image without a file ending, an img tag with a non-base64 src attribute, etc.) - WDYT?